include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -1610,6 +1618,14 @@ namespace PETScWrappers
     prepare_action(VectorOperation::insert);
   }
 
+  inline const MPI_Comm &
+  MatrixBase::get_mpi_communicator() const
+  {
+    static MPI_Comm comm;
+    PetscObjectGetComm(reinterpret_cast<PetscObject>(matrix), &comm);
```

Comment:
Could we just set up the communicator in the ctor? I would prefer to not use a static variable here.

There shouldn't be a circumstance in which the communicator changes between calls, unless there is a call to `reinit()` or some similar function

Reply:
The return of the reference is a remnant of having a communicator stored inside the PETSc classes. I don't know why you kept it there, but it is useless. Once all my changes are incorporated, we can make this function return a`MPI_Comm` instead of `MPI_Comm&`.  I don't understand what you mean by "setup the communicator in the constructor". Can you elaborate?

Reply:
I meant that we could add `MPI_Comm` as a class member and just return a reference to that instead.

Reply:
Why do you want to complicate the code maintainability when you can always return the comm stored in the PETSc object? In some sense, the comm is already stored in the class, since it is stored in the PETSc object. The point is why you want a reference to a MPI_Comm as return value

Reply:
That's a better choice - I don't mind if we change the API to return the MPI communicator by value (like what our Trilinos vectors do) instead of using the static variable, I just want to avoid the static variable.

Reply:
Ok. What about making this change in a final pull request after all other PRs are merged?

Reply:
Actually, if we `clear` the matrix we currently replace the `Mat` object with an empty serial matrix, initialized with `MPI_COMM_SELF`. In principle, you are also allowed to change the `Mat` object stored in this class to use a different communicator (directly using PETSc).

In the way this is implemented, we always ask the underlying mat object. Maintaining the `comm` static here, and not directly visible as a member, has the advantage that you are forced to access it through the `get_mpi_communicator` and this guarantees that you are aligned with underlying `Mat` object.

Reply:
I think we should use the accessor either way. @luca-heltai do you agree with making this return by value?

Reply:
Yes. But let's do this at the end of the string of PRs. I'm opening an issue for this.